src/Components/MainMenu/MainMenu.js

```diff
@@ -169,7 +178,11 @@ class MainMenu extends React.Component {
       >
         <div className="main-menu-top">
           {this.renderHeader()}
-          <ul className="main-menu-items">
+          <ul
+            className={classNames('main-menu-items', {
+              nested: this.isDeeplyNested(menu),
```

Comment:
Don't do this here, this will call that fun ever render even if the props don't change. I would make this calculation on the ctor and when it receives props and set a property and use that here.